pkg/controller/deployment/deployment_controller.go

```diff
@@ -560,8 +556,48 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 
 	// Sync deployment status
 	return dc.syncDeploymentStatus(allRCs, newRC, deployment)
+}
 
-	// TODO: raise an event, neither scaled up nor down.
+func (dc *DeploymentController) syncPausedDeployment(deployment extensions.Deployment) error {
+	newRC, oldRCs, err := dc.getNewRCAndOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+	if len(oldRCs) == 0 {
+		// No old rcs with pods; scale new rc to the full count.
+		_, err := dc.scaleRCAndRecordEvent(newRC, deployment.Spec.Replicas, deployment)
+		return err
+	}
+	allRCs := append(oldRCs, newRC)
+	// Sort
+	sort.Sort(controller.ByAvailableReplicas(allRCs))
+	maxSurge, isPercent, err := util.GetIntOrPercentValue(&deployment.Spec.Strategy.RollingUpdate.MaxSurge)
+	if err != nil {
+		return fmt.Errorf("invalid value for MaxSurge: %v", err)
+	}
+	if isPercent {
+		maxSurge = util.GetValueFromPercent(maxSurge, deployment.Spec.Replicas)
+	}
+	// Number of pods that can be either added or removed from the total pod count.
+	allowed := deployment.Spec.Replicas + maxSurge - deployment.Status.AvailableReplicas
+	if allowed == 0 {
+		return nil
+	}
+	// Scale proportionally
+	for i := range allRCs {
+		rc := allRCs[i]
+		// This can be either positive (scale up) or negative (scale down). Rounds to the integer
+		// value.
+		proportion := int((rc.Status.Replicas / deployment.Status.AvailableReplicas) * allowed)
```

Comment:
So if I understand this correctly, we want proportional scaling when there are more than the two replica sets we have in the usual rollout case?